src/lib.rs

```diff
@@ -563,25 +579,24 @@ impl<M: Manager> Pool<M> {
             }
         }
 
-        internals.num_open += 1;
```

Comment:
I think it was a bug there! There `num_open` became incorrect when the `connect` action below was failed.